Clang format  and refactor Game.h/c

ricfehr3 opened the pull request:
Refactored `game.c` quite a bit. Mostly extracting all the file scoped functions into `static` functions. No real effect on the size of the ROM either, the only real effect is aesthetic.

There is no code change here. Just formatting and moving stuff around.

ricfehr3 commented:
@MeirGavish That's definitely relevant, haha. I'm not going to hunt places down to put that prefix right now, but you're right, there is room for more discussion. As for this PR, I'm thinking we leave everything static as long as it's within the scope of the file. Once it leaves the file, that PR will properly name the functions and set scope.

I do think that the `s_` could be used for static functions instead of the `_`, but because we are planning to gut this file, I won't do that here.

ricfehr3 commented:
> Working on a review for this, the file is big so it's taking me a bit, I'll try to finish it soon.

No worries at all. I'm having a chill holiday, so no rush :)

MeirGavish commented:
> No worries at all. I'm having a chill holiday, so no rush :) 

Happy Thanksgiving!

MeirGavish reviewed (changes requested):
Submitting the review so you're not waiting a long time for it, I still need to give the file another pass though...

ricfehr3 commented on source/game.c:
```diff
@@ -2844,142 +3651,59 @@ static void game_round_end_print_interest_reward(int interest_y_offset)
+                   interest_y * TILE_SIZE,
+                   TTE_YELLOW_PB,
+                   _interest_reward,
+                   TTE_WHITE_PB);
```
Yeah, clang-format seems to be the most awesome, but limiting thing, haha. Still, I think it's a good addition but these tiny things are bugging me too.

ricfehr3 commented on source/game.c:
```diff
@@ -49,190 +155,425 @@ typedef enum
+static const BG_POINT HAND_START_POS = {120, 90};
+static const BG_POINT HAND_PLAY_POS = {120, 70};
+static const BG_POINT MAIN_MENU_ACE_T = {88, 26};
+// Naming the stage where cards return from the discard pile to the deck "undiscard"
```
Put it in again, I merged main back in an it looks like I started on a previous commit without that #define? Not sure, but it's here now.

ricfehr3 commented on source/game.c:
```diff
@@ -49,190 +155,425 @@ typedef enum
+static void card_in_hand_loop_handle_discard_and_shuffling(int card_idx,
+                                                           FIXED* hand_x,
+                                                           FIXED* hand_y,
+                                                           bool* break_loop);
```
I like this. I got it working and I'll push it soon and you can decide if that works for you. It does look nicer to me.

ricfehr3 commented on source/game.c:
```diff
@@ -2844,142 +3651,59 @@ static void game_round_end_print_interest_reward(int interest_y_offset)
-#define NEXT_ROUND_BTN_SEL_X 0
-
-#define REROLL_BTN_FRAME_PAL_IDX        7
-#define REROLL_BTN_PAL_IDX              3
```
Yeah, I'm thinking we keep it consistent for now. It shouldn't be too difficult to separate when we get to it. However nothing is set in stone, just let me know if you want to switch it back.

ricfehr3 commented:
@MeirGavish This one should be good. I addressed all the comments. But there are now more changes here so understandable if you need more time, and sorry for the massive amount of changes

ricfehr3 commented:
@MeirGavish Okay, after looking at the other clang format PRs, I'd like to merge this one first because the addition of that [request you had](https://github.com/cellos51/balatro-gba/pull/259#discussion_r2559875012) really cleaned everything up. I'd like to apply it to the other PRs, but this one contains a lot of the changes required already.

MeirGavish commented:
> > > @MeirGavish Okay, after looking at the other clang format PRs, I'd like to merge this one first because the addition of that [request you had](https://github.com/cellos51/balatro-gba/pull/259#discussion_r2559875012) really cleaned everything up. I'd like to apply it to the other PRs, but this one contains a lot of the changes required already.
> > 
> > 
> > @ricfehr3 But not before the truncate scores PR #269, right? I'll try to give this another pass today.
> 
> Yeah, that's correct. We should merge #269 before this for sure. This one should be the first of the remaining clang-formats though. And Thanks!

Thanks! Sure! I meant to do it yesterday but didn't get to it, I was pretty sleep deprived.

MeirGavish reviewed (changes requested):
Looks good, just restore a deleted comment, and maybe add a bit more organization to the forward declarations if possible (I guess that's something I missed in the previous review)

ricfehr3 commented on source/game.c:
```diff
@@ -48,145 +150,392 @@ typedef enum
 // Thanks!
 // https://github.com/cellos51/balatro-gba/issues/137#issuecomment-3322485129
-static void _noop() {}
+static void noop()
```
Yeah, this is a niche one. I did add it to the code style guide, but I haven't been enforcing it 100%. I'm considering removing it. It really isn't applicable. While it is correctly stated that the standard says that, we are supporting one compiler, which does catch this error. Buuuuuut, after some testing I think we should stick to it, it's a c99 standard. I'm not sure what we auto default too, but supporting c99 is probably good. Maybe we should add a standard in the makefile to ensure it's right too? 

That all being said, adding this suggestion.
